new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1148 @@
+    Path dstDsOverrideFile = Paths.get(WORK_DIR, "jdbc-JdbcTestDataSource-1.xml");
+    String tempString = assertDoesNotThrow(()
+        -> Files.readString(srcDsOverrideFile).replaceAll("JDBC_URL", dsUrl2));
+    assertDoesNotThrow(()
+        -> Files.write(dstDsOverrideFile, tempString.getBytes(StandardCharsets.UTF_8)));
+
+    List<Path> overrideFiles = new ArrayList<>();
+    overrideFiles.add(dstDsOverrideFile);
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset1/config.xml"));
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset1/version.txt"));
+
+    //create config override map
+    createConfigMapFromFiles(overridecm, overrideFiles, domainNamespace);
+
+    //patch the domain resource with overridesConfigMap, secrets and introspectVersion
+    patchStr
+        = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/overridesConfigMap\", \"value\": \"" + overridecm + "\"},"
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/secrets\", \"value\": [\"" + dsSecret + "\"]  },"
+        + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"3\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    verifyIntrospectorRuns();
+    verifyPodsStateNotChanged();
+
+    try {
+      //wait for a minute to see if the overrides are not applied
+      TimeUnit.MINUTES.sleep(1);
+    } catch (InterruptedException ex) {
+      //ignore
+    }
+
+    //verify the overrides are not applied
+    verifyConfigXMLOverride(false);
+    verifyResourceJDBC0Override(false);
+
+    //restart domain for the distributionstrategy to take effect
+    restartDomain();
+
+    //verify on restart the overrides are applied
+    verifyConfigXMLOverride(true);
+    verifyResourceJDBC1Override(true);
+
+    //cleanup secret
+    deleteSecret(dsSecret, domainNamespace);
+    patchStr
+        = "["
+        + "{\"op\": \"remove\", \"path\": \"/spec/configuration/secrets\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    patch = new V1Patch(patchStr);
+    patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH);
+  }
+
+  /**
+   * Test patching the domain with values for /spec/configuration/overrideDistributionStrategy field anything other than
+   * DYNAMIC or ON_RESTART fails.
+   *
+   * <p>Test tries to set the above field to RESTART and asserts the patching fails.
+   */
+  @Disabled
+  @Order(4)
+  @Test
+  @DisplayName("Test invalid overrideDistributionStrategy value RESTART")
+  public void testOverrideNegative() {
+
+    //patching the domain with /spec/configuration/overrideDistributionStrategy: RESTART
```

Comment:
Is it not a unit test candidate ?

Reply:
may be, but it doesn't take much time to run this test, so covering it.

Reply:
ok